src/imap.rs

```diff
@@ -2468,4 +2479,27 @@ mod tests {
         }
         Ok(())
     }
+
+    #[async_std::test]
+    async fn test_get_imap_search_command() -> Result<()> {
+        let t = TestContext::new_alice().await;
+        assert_eq!(
+            get_imap_self_sent_search_command(&t.ctx).await?,
+            "FROM alice@example.org"
+        );
+
+        t.ctx.set_primary_self_addr("alice@another.com").await?;
+        assert_eq!(
+            get_imap_self_sent_search_command(&t.ctx).await?,
+            "OR (FROM alice@another.com) (FROM alice@example.org)"
+        );
+
+        t.ctx.set_primary_self_addr("alice@third.com").await?;
+        assert_eq!(
+            get_imap_self_sent_search_command(&t.ctx).await?,
+            "OR (OR (FROM alice@third.com) (FROM alice@another.com)) (FROM alice@example.org)"
+        );
```

Comment:
I came up with this from reading IETF docs but can not guarantee this actually works ;)